### PR TITLE
Document why using MPI_COMM_WORLD is ok in one place.

### PR DESCRIPTION
**Pull request by bangerth**
This addresses the other part of #6016.

### CODE REVIEW

File: source/lac/scalapack.cc [hunk 1]
```diff
@@ -407,8 +407,18 @@ ScaLAPACKMatrix<NumberType>::copy_to (ScaLAPACKMatrix<NumberType> &dest) const
...
-      // to create a communicator representing the union of the source and destination MPI communicator
-      // we need a communicator containing all  desired processes --> use MPI_COMM_WORLD
+      
+      // to create a communicator representing the union of the source
+      // and destination MPI communicator we need a communicator that
+      // is guaranteed to contain all desired processes -- i.e.,
+      // MPI_COMM_WORLD. on the other hand, as documented in the MPI
+      // standard, MPI_Comm_create_group is not collective on all
+      // processes in the first argument, but instead is collective on
+      // only those processes listed in the group. in other words,
```
Comment (davydden) [inline source/lac/scalapack.cc]: In 

(capital)

**Comment by davydden** [comment]
@BenBrands FYI

File: source/lac/scalapack.cc (hunk 1 above)
Comment (bangerth) [reply source/lac/scalapack.cc]: ah, we do this everywhere in the documentation that we start sentences with lower-case letters...

Comment (davydden) [reply source/lac/scalapack.cc]: hm, never noticed this, nevermind then

**Comment by davydden** [comment]
@bangerth please fix indentation.

File: source/lac/scalapack.cc (hunk 1 above)
Comment (tjhei) [reply source/lac/scalapack.cc]: ... and I have been trying to fight this for years but @bangerth keeps adding more and more of these. ;-)

Comment (bangerth) [reply source/lac/scalapack.cc]: Alright alright, I'll change my ways in the future.

**Comment by bangerth** [comment]
Properly indented patch now here.

**Pull request merged**